platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/component/ComponentWiring.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring.component;
+
+import com.swirlds.common.wiring.component.internal.WireBindInfo;
+import com.swirlds.common.wiring.component.internal.WiringComponentProxy;
+import com.swirlds.common.wiring.schedulers.TaskScheduler;
+import com.swirlds.common.wiring.wires.input.BindableInputWire;
+import com.swirlds.common.wiring.wires.input.InputWire;
+import com.swirlds.common.wiring.wires.output.OutputWire;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+
+/**
+ * Builds and manages input/output wires for a component.
+ *
+ * @param <COMPONENT_TYPE> the type of the component
+ * @param <OUTPUT_TYPE>    the output type of the component
+ */
+public class ComponentWiring<COMPONENT_TYPE, OUTPUT_TYPE> {
+
+    private final TaskScheduler<OUTPUT_TYPE> scheduler;
+
+    private final WiringComponentProxy proxy = new WiringComponentProxy();
+    private final COMPONENT_TYPE proxyComponent;
+
+    private COMPONENT_TYPE component;
+
+    private final Map<Method, WireBindInfo> bindInfo = new HashMap<>();
+    private final Map<Method, BindableInputWire<Object, Object>> inputWires = new HashMap<>();
+
+    /**
+     * Create a new component wiring.
+     *
+     * @param clazz     the class of the component
```

Comment:
Maybe we should specify that this should be an interface, and throw immediately if its not

Reply:
check added